crates/driver/src/logic/eth/eip712.rs

```diff
@@ -9,7 +9,7 @@
 pub struct DomainSeparator(pub [u8; 32]);
 
 impl DomainSeparator {
-    pub fn new(chain_id: super::ChainId, verifying_contract: super::Contract) -> Self {
+    pub fn new(chain_id: super::ChainId, verifying_contract: super::ContractAddress) -> Self {
```

Comment:
As a side note, this is not a general `DomainSeparator`, but a CoW Protocol specific one.